src/NUnitFramework/framework/Internal/ParamAttributeTypeConversions.cs

```diff
@@ -104,22 +115,33 @@ public static bool TryConvert(object value, Type targetType, out object converte
 
             bool convert = false;
 
-            if (targetType == typeof(short) || targetType == typeof(byte) || targetType == typeof(sbyte))
+            if (targetType == typeof(short) || targetType == typeof(byte) || targetType == typeof(sbyte) || targetType == typeof(long?) ||
+                targetType == typeof(short?) || targetType == typeof(byte?) || targetType == typeof(sbyte?) || targetType == typeof(double?))
             {
                 convert = value is int;
             }
-            else if (targetType == typeof(decimal))
+            else if (targetType == typeof(decimal) || targetType == typeof(decimal?))
             {
                 convert = value is double || value is string || value is int;
             }
-            else if (targetType == typeof(DateTime) || targetType == typeof(TimeSpan))
+            else if (targetType == typeof(DateTime) || targetType == typeof(DateTime?))
             {
                 convert = value is string;
             }
 
             if (convert)
             {
-                convertedValue = System.Convert.ChangeType(value, targetType, System.Globalization.CultureInfo.InvariantCulture);
+                Type convertTo = targetType.GetTypeInfo().IsGenericType && targetType.GetGenericTypeDefinition() == typeof(Nullable<>) ?
+                    targetType.GetGenericArguments()[0] : targetType;
+
+                convertedValue = System.Convert.ChangeType(value, convertTo, CultureInfo.InvariantCulture);
+                return true;
+            }
+
+            var converter = TypeDescriptor.GetConverter(targetType);
+            if (converter.CanConvertFrom(value.GetType()))
```

Comment:
I'd appreciate some input on this especially. Is it better to use `CanConvertFrom`, or to have a specific list of what should be supported (like this: https://github.com/nunit/nunit/pull/3098/files#diff-64bcafa7925c912b792f366a7971facfR123)

Reply:
I don't see the problem this would create as written, but would appreciate @nunit/framework-team's thoughts on this!

Reply:
I'd like to support all types this way, too.

Should GetConverter be given `Nullable.GetUnderlyingType(targetType)` if it's not null?